.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,50 @@
+name: Benchmark
+
+on:
+  push:
+    branches:
+      - main
+      - ci-benchmark-workflow  # only temporary, not for commit
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+
+jobs:
+  build-and-bench:
+    name: Build and Benchmark on big-iron8
+    runs-on: self-hosted
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # the secret runner token needs to be referenced by at least one
+      # step in order to be sent along with the job in the first place
+      - name: Check Runner Token Access
+        run: |
+          if [ -z "$TOKEN" ]; then
+            echo "Unable to access runner token!"
+            exit 1
+          fi
+        env:
+          TOKEN: ${{ secrets.RUNNER_TOKEN }}
```

Comment:
There is also a check on the Github runner, right?

Reply:
Maybe this would be worth to document. People could be scared to see this "check", because they do not know about your forked GitHub runner :)

Reply:
Good point. Now that you mentioned it I realize that referencing the token in a "check" is somewhat misleading. I have now removed the entire step specific to the token and moved it into the job-scoped environment instead. This way it should be more obvious that we use it in a job-wide fashion. I have also expanded the existing comment with a reference to our forked GitHub runner. This should address your points I think.

PTAL.

Reply:
Thanks! It is much better now with your change. Even newcomers should easily understand that.